lms/models/json_settings.py

```diff
@@ -3,24 +3,52 @@
 from typing import Any, Optional, Tuple
 
 import sqlalchemy as sa
+from pyramid.settings import asbool
 from sqlalchemy.ext.mutable import MutableDict
 from sqlalchemy.orm import InstrumentedAttribute
 
 
+
+class _JSONSettingType:
+    def format_for_storage(self, value):
+        raise NotImplementedError("You cannot store this type directly")
+
+
+class _StringOrNone(_JSONSettingType):
+    def format_for_storage(self, value):
+        if value:
+            value = value.strip()
+
+        if not value:
+            return None
+
+        return value
+
+
+class _Boolean(_JSONSettingType):
+    format_for_storage = asbool
+
+
+class _AESSecret(_JSONSettingType):
+    ...
+
+
 @dataclass
 class JSONSetting:
     """Describe a permitted field in a JSONSettings object."""
 
-    # Helper to declare settings as secret. This can be used with format
-    AES_SECRET = object()
+    class Types:
+        STRING = _StringOrNone
+        BOOLEAN = _Boolean
+        AES_SECRET = _AESSecret
```

Comment:
I honestly can't tell if this is better or not. 

In the pros camp:

* The objects here have spaces for other functions, should we ever need them
* It makes a clear distinction between the format function and the type definition

In the cons camp:

* YAGNI - In the event we ever need more than one function we can add it
* It's honestly just a fancy way of hiding what we were already doing
* We could have just done `from pyramid.settings import asbool as BOOLEAN` with much the same effect